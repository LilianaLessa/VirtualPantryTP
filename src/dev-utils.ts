// eslint-disable-next-line import/no-extraneous-dependencies
import { faker } from "@faker-js/faker";
// eslint-disable-next-line import/no-extraneous-dependencies
import { v4 as uuidv4 } from "uuid";
import Product from "./features/products/classes/product.class";
import { IProduct } from "./features/products/interfaces/product.interface";

export const createMockProduct = (): IProduct =>
  new Product(uuidv4(), faker.commerce.productName());
