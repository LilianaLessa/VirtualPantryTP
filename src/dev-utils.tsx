import { faker } from "@faker-js/faker";
// eslint-disable-next-line import/no-extraneous-dependencies
import { v4 as uuidv4 } from "uuid";
import { Text, View } from "react-native";
import React from "react";
import Product from "./features/products/classes/product.class";
import Pantry from "./features/pantries/classes/pantry.class";
import ShoppingList from "./features/shoppingList/classes/shopping-list.class";
import Group from "./features/group/classes/group.class";
import UserInGroup, {
  UseInGroupAcceptanceState,
} from "./features/group/classes/user-in-group.class";
import Notification from "./features/notification/classes/notification.class";
import StoredProduct from "./features/products/classes/stored.product";
import ShoppingListItem from "./features/shoppingList/classes/shopping-list-item.class";

export function getStackTraceAsString(error: Error): string {
  if (error.stack) {
    return error.stack
      .split("\n")
      .slice(2)
      .map((line) => line.replace(/\s+at\s+/, ""))
      .join("\n");
  }

  return "";
}

export function createMockShoppingListItem(): ShoppingListItem {
  return new ShoppingListItem(uuidv4(), "");
}

export function createMockShoppingList(): ShoppingList {
  return new ShoppingList(uuidv4());
}

export const createMockProduct = (name?: string): Product =>
  new Product(uuidv4(), "", name ?? faker.commerce.productName());

export const createMockShoppingLists = (): ShoppingList =>
  new ShoppingList(uuidv4(), faker.word.verb());

export const createMockPantry = (name?: string): Pantry =>
  new Pantry(uuidv4(), name ?? faker.name.firstName());

export function ScreenPlaceHolder({ route }: { route: { params: any } }) {
  const { screenName } = route.params;
  return (
    <View>
      <Text>{screenName} Screen Placeholder</Text>
    </View>
  );
}

export function createMockStoredProduct(): StoredProduct {
  return new StoredProduct(uuidv4(), "");
}

export function createMockGroup(): Group {
  return new Group(uuidv4(), "mock group", "ownerUid");
}

export function createMockUserInGroup(group: Group): UserInGroup {
  return new UserInGroup(
    uuidv4(),
    "",
    group.uuid,
    "",
    false,
    false,
    UseInGroupAcceptanceState.PENDING
  );
}

export function createMockNotification(): Notification {
  return new Notification(
    "UUID",
    undefined,
    undefined,
    false,
    new Date().toLocaleDateString(),
    ""
  );
}
