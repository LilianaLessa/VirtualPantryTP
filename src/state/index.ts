import * as actionCreators from "./action-creators";

export * from "./store";
export * from "./reducers";

/**
 * to solve
 * Export namespace should be first transformed by `@babel/plugin-proposal-export-namespace-from`.
 * @see https://stackoverflow.com/questions/61718768/how-can-i-export-as-namespace-in-babel
 */
export { actionCreators };
