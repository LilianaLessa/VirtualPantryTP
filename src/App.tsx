import * as React from "react";
import { Provider } from "react-redux";
import { NavigationContainer, useNavigation } from "@react-navigation/native";

import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { ActivityIndicator, MD2Colors, Snackbar } from "react-native-paper";
import { SafeAreaProvider } from "react-native-safe-area-context";
import Modal from "react-native-modal";
import { TouchableOpacity, View } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { store } from "./state";
import { useTypedSelector } from "./hooks/useTypedSelector";
import { useActions } from "./hooks/useActions";
import HomeScreen from "./features/home/screens/home.screen";
import { ScreenPlaceHolder } from "./dev-utils";
import {
  EditProductScreen,
  EditProductScreenParams,
} from "./features/products/screens/edit-product.screen";

import { BarCodeScannerContextProvider } from "./services/barCodeScanner/barCodeScanner.context";
import {
  ProductSearchResultScreen,
  ProductSearchResultsScreenParams,
} from "./features/products/screens/product-search-results.screen";
import {
  BarCodeScanScreenRouteName,
  EditProductScreenRouteName,
  ProductScreenRouteName,
  ProductSearchResultScreenRouteName,
} from "./infrastructure/navigation/route-names";
import ProductScreen from "./features/products/screens/product.screen";
import BarCodeScanScreen from "./features/products/screens/barcode-scan.screen";

function LoadingModal() {
  const { fetchingData } = useTypedSelector((state) => state.apiActivity);
  return (
    <Modal isVisible={fetchingData}>
      <ActivityIndicator animating color={MD2Colors.red800} size="large" />
    </Modal>
  );
}

function ErrorSnack() {
  const { errorVisible, errorMessage } = useTypedSelector(
    // eslint-disable-next-line comma-dangle
    (state) => state.messageSnackbar
  );
  const { hideErrorSnack } = useActions();

  return (
    <Snackbar
      style={{
        backgroundColor: "#B22D1D",
      }}
      visible={errorVisible}
      onDismiss={() => hideErrorSnack()}
      action={{
        label: "Dismiss",
        onPress: () => hideErrorSnack(),
      }}
      duration={7000}
    >
      {errorMessage}
    </Snackbar>
  );
}

function InfoSnack() {
  const { infoVisible, infoMessage } = useTypedSelector(
    // eslint-disable-next-line comma-dangle
    (state) => state.messageSnackbar
  );
  const { hideInfoSnack } = useActions();

  return (
    <Snackbar
      visible={infoVisible}
      onDismiss={() => hideInfoSnack()}
      action={{
        label: "Dismiss",
        onPress: () => hideInfoSnack(),
      }}
      duration={3000}
    >
      {infoMessage}
    </Snackbar>
  );
}

function HeaderLeftActions() {
  const navigation = useNavigation();

  return (
    <View>
      <TouchableOpacity
        onPress={() => navigation.navigate("ConfigurationsScreen" as never)}
      >
        <MaterialCommunityIcons name="menu" size={24} color="black" />
      </TouchableOpacity>
    </View>
  );
}

function HeaderRightActions() {
  const navigation = useNavigation();

  return (
    <View>
      <TouchableOpacity
        onPress={() => navigation.navigate("NotificationsScreen" as never)}
      >
        <MaterialCommunityIcons name="bell" size={24} color="black" />
      </TouchableOpacity>
    </View>
  );
}

type RootStackParamList =
  | { Home: undefined }
  | { Products: undefined }
  | { BarCodeScanScreen: undefined }
  | { NotificationsScreen: { screenName: string } }
  | { ConfigurationsScreen: { screenName: string } }
  | { Pantries: { screenName: string } }
  | { ShoppingLists: { screenName: string } }
  | { Groups: { screenName: string } }
  | ProductSearchResultsScreenParams
  | EditProductScreenParams;

const Stack = createNativeStackNavigator<RootStackParamList>();

function App() {
  return (
    <SafeAreaProvider>
      <NavigationContainer>
        <Provider store={store}>
          <BarCodeScannerContextProvider>
            <Stack.Navigator initialRouteName="Home">
              <Stack.Screen
                name="Home"
                component={HomeScreen}
                options={{
                  headerLeft: () => <HeaderLeftActions />,
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name={ProductScreenRouteName as never}
                component={ProductScreen}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name={BarCodeScanScreenRouteName as never}
                component={BarCodeScanScreen}
              />
              <Stack.Screen
                name="NotificationsScreen"
                component={ScreenPlaceHolder}
                initialParams={{ screenName: "Notifications" }}
              />
              <Stack.Screen
                name="ConfigurationsScreen"
                component={ScreenPlaceHolder}
                initialParams={{ screenName: "Configurations" }}
              />
              <Stack.Screen
                name="Pantries"
                component={ScreenPlaceHolder}
                initialParams={{ screenName: "Pantries" }}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name="ShoppingLists"
                component={ScreenPlaceHolder}
                initialParams={{ screenName: "Shopping Lists" }}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name="Groups"
                component={ScreenPlaceHolder}
                initialParams={{ screenName: "Groups" }}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name={EditProductScreenRouteName as never}
                component={EditProductScreen}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
              <Stack.Screen
                name={ProductSearchResultScreenRouteName as never}
                component={ProductSearchResultScreen}
                options={{
                  headerRight: () => <HeaderRightActions />,
                }}
              />
            </Stack.Navigator>
            <LoadingModal />
            <ErrorSnack />
            <InfoSnack />
          </BarCodeScannerContextProvider>
        </Provider>
      </NavigationContainer>
    </SafeAreaProvider>
  );
}

export default App;
