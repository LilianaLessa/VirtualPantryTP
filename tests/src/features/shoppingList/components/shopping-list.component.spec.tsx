// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import * as React from "react";
import * as renderer from "react-test-renderer";
import { createMockShoppingList } from "../../../../../src/dev-utils";
import ShoppingList from "../../../../../src/features/shoppingList/components/shopping-list.component";

describe("App", () => {
  it("snapshot test", () => {
    const tree = renderer
      .create(
        <ShoppingList
          shoppingList={createMockShoppingList()}
          removeItemCallback={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
