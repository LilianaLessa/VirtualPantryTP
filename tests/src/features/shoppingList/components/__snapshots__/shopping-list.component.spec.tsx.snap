// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "marginBottom": 5,
      "opacity": 1,
      "paddingLeft": 10,
      "paddingRight": 10,
    }
  }
>
  <View
    className="sc-beqWaB frbneG"
  >
    <View
      className="sc-gueYoa fxeKqo"
    >
      <Text />
      <Text
        className="sc-eDDNvR btIXrk"
      />
    </View>
    <View
      className="sc-jTrPJq hGlsa-d"
    >
      <Text>
        0/0
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    accessibilityRole="progressbar"
    accessibilityState={
      {
        "busy": true,
      }
    }
    accessibilityValue={
      {
        "max": 100,
        "min": 0,
        "now": 0,
      }
    }
    accessible={true}
    onLayout={[Function]}
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(231, 224, 236, 1)",
          "height": 4,
          "opacity": 0,
          "overflow": "hidden",
        }
      }
    />
  </View>
</View>
`;
